src/engine/runtime.js

```diff
@@ -812,18 +825,16 @@ class Runtime extends EventEmitter {
      * @private
      */
     _refreshExtensionPrimitives (extensionInfo) {
-        let extensionBlocks = [];
         for (const categoryInfo of this._blockInfo) {
             if (extensionInfo.id === categoryInfo.id) {
                 categoryInfo.name = maybeFormatMessage(extensionInfo.name);
                 categoryInfo.blocks = [];
                 categoryInfo.menus = [];
                 this._fillExtensionCategory(categoryInfo, extensionInfo);
-                extensionBlocks = extensionBlocks.concat(categoryInfo.blocks, categoryInfo.menus);
+
+                this.emit(Runtime.BLOCKSINFO_UPDATE, categoryInfo);
```

Comment:
Documenting a conversation from offline...
It looks like this change would make this line run multiple times per call to `_refreshExtensionPrimitives` but in practice, this change is the same as before, this emit runs only once per call to `_refreshExtensionPrimitives` because there should only ever be one category in `this._blockInfo` which matches the given extension id.

We should refactor this in the future to prevent confusion.

Reply:
Refactored in fd776025e5fe2cfd2b2d5f7fa004ea94985b1447